File: backend/cadastros/src/main/java/com/ecsolutions/cadastros/model/dtos/project/ProjectRequestDto.java
package com.ecsolutions.cadastros.model.dtos.project;

import com.ecsolutions.cadastros.model.enums.SimpleStatusEnum;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProjectRequestDto {

        private UUID creatorId;

        private UUID clientId;

        private String title;

        private String description;
}

File: frontend/src/app/_models/projects/project-request.dto.ts
import { z } from "zod";

export const ProjectRequestSchema = z.object({
    creatorId: z.string().uuid(),
    clientId: z.string().uuid(),
    title: z.string(),
    description: z.string().optional().nullable(),
})

export type ProjectRequestDto = z.infer<typeof ProjectRequestSchema>;